httplib.h

```diff
@@ -435,6 +435,12 @@ struct scope_exit {
 
 } // namespace detail
 
+enum SSLVerifierResponse {
+  Verified, // connection certificate is verified and accepted
+  CheckAgain, // use the built-in certificate checker again
+  Declined // connection certificate was process but is declined
+};
```

Comment:
Might I suggest different names:

```cpp
enum SSLVerifierResponse {
  NoDecisionMade,        // no decision has been made, use built-in certificate verifier
  CertificateAccepted,   // connection certificate is verified and accepted
  CertificateRejected    // connection certificate was processed but is rejected
};
```

Also, order `NoDecisionMade` to the top. It makes more sense to be the default enum value and is certainly less dangerous than `CertificateAccepted`.

Just two cents from a random contributor...

Reply:
Proposed change I think is not bad, will update PR.

But since I first posted #2061 and asked if I should publish PR and got an "OK" from @yhirose I think it should be OK to be proposing public API change, although not a major one IMHO; final decision is still his.